src/tabs/serverAdmins/ManagePermissionsDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import RX = require("reactxp");
+import Styles = require("../../styles/main");
+import {CheckBox} from "../../components/CheckBox";
+import {OkCancelButtonBar} from "../../OkCancelButtonBar";
+import {AdminPermissions} from "./ServerAdminsTabState";
+
+interface ManagePermissionsDialogProps {
```

Comment:
In my understanding there is duplicate information inside this structure - the ID is stored both in `adminId` and in `adminPermissions.id`. So I'd either
- remove the `adminId` field, or
- change the type of `adminPermissions` from `AdminPermissions` to `Permissions`, which - if I understand correctly - only encapsulates the actual permission settings and not the ID

and obviously update the code which uses this structure accordingly.